src/public/providers.ts

```diff
@@ -4,82 +4,99 @@ const providerList: HTMLElement = helper.getElement('.providerList')
 type Provider = {
   id?: string
   name: string
-  api: string
-  keys: string[]
+  service: string
+  keys: Array<ProviderKey>
+  // eslint-disable-next-line
+  domains: Array<any>
 }
 
-/* HardCoded dns object to display mock UI */
-const temp = {
-  dnsGd: {
-    name: 'GoDaddy',
-    api: 'https://api.godaddy.com',
-    keys: ['GD_Key', 'GD_Secret']
-  },
-  dnsPdns: {
-    name: 'PowerDNS',
-    api: 'https://api.powerdns.com',
-    keys: ['PDNS_Url', 'PDNS_ServerId', 'PDNS_Token', 'PDNS_Ttl']
+type ProviderKey = {
+  id?: string
+  key: string
+  service: string
+  value: string
+}
+
+class DomainElement {
+  // eslint-disable-next-line
+  constructor(domainObj: any, container: HTMLElement) {
+    const domainElement = document.createElement('div')
+    domainElement.innerHTML = `
+      <span class="domainElement">${domainObj.domain}</span>
+      <button type="button" class="btn btn-primary setupButton">Setup</button>
+    `
+    container.appendChild(domainElement)
```

Comment:
if you are doing `DomainElement`, you might as well just add an `onclick` binding to send a `POST` to `/api/admin/sslCert`